wa/workloads/uibench/__init__.py

```diff
@@ -47,6 +49,27 @@ class Uibench(ApkWorkload):
                   """),
     ]
 
-    def run(self, context):
-        super(Uibench, self).run(context)
-        self.target.sleep(self.duration)
+    def __init__(self, target, **kwargs):
+        super(Uibench, self).__init__(target, **kwargs)
+        if self.activity:
+            activities = self.activity.split()
+            self.activity = ''
+            for activity in activities:
+                if '.' not in activity:
+                    # If we're receiving just the activity name, it's taken relative to
+                    # the package namespace:
+                    self.activity += ' .' + activity
+                else:
+                    self.activity += ' ' + activity
+
+    def setup(self, context):
+        if self.activity:
+            activities = self.activity.split()
+            for activity in activities:
+                self.apk._activity = activity
+                self.logger.info("Starting {} for {} seconds".format(activity, self.duration))
+                super(Uibench, self).setup(context)
+                self.target.sleep(self.duration)
```

Comment:
It has always been like this. But the difference now is that we sleep in the setup instead of run, which is your point. Let me try to follow on with your suggestion which should address this too. Thanks.